addons/payment_razorpay/tests/test_payment_provider.py

```diff
@@ -24,3 +25,27 @@ def test_signature_calculation_for_redirect_data(self):
         self.assertEqual(
             calculated_signature, '437b72e4e87362a39951b44487cf698410b074afdbed19ec44fffd32d2f863f3'
         )
+
+    def test_credentials_constraints(self):
```

Comment:
Same, and it's even more important with tests, so that:
- A test function doesn't quickly become a junkyard where everyone slaps a new assertSomething for every new bug.
- One can quickly figure out the issue from the test logs: `test_stuff` failed VS `test_confirming_order_cancels_draft_transactions` failed, for example.
```suggestion
    def test_prevent_enabling_if_credentials_are_not_set(self):
```